broadlink/climate.py

```diff
@@ -201,3 +201,282 @@ def set_schedule(self, weekday: List[dict], weekend: List[dict]) -> None:
+        data['set_temp'] = (payload[0x0c] >> 3) + 8 + (0.0 if ((payload[0xe] & 0b10000000) == 0) else 0.5)
+
+        swing_v = payload[0x0c] & 0b111
+        swing_h = (payload[0x0d] & 0b11100000) >> 5
+        if (swing_h == 0b111):
+            data['swing_h'] = 'OFF'
+        elif (swing_h == 0b111):
+            data['swing_h'] = 'OFF'
+        else:
+            data['swing_h'] = 'unrecognized value'
+        
+        if (swing_v == 0b111):
+            data['swing_v'] = 'OFF'
+        elif (swing_v == 0b000):
+            data['swing_v'] = 'ON'
+        elif (swing_v >= 0 and swing_v <=5):
+            data['swing_v'] = swing_v
+        else:
+            data['swing_v'] = 'unrecognized value'
+
+        mode = payload[0x11] >> 3 << 3
+        if mode == 0x00:
+            data['mode'] = 'A'
+        elif mode == 0x20:
+            data['mode'] = 'C'
+        elif mode == 0x40:
+            data['mode'] = 'D'
+        elif mode == 0x80:
+            data['mode'] = 'H'
+        elif mode == 0xc0:
+            data['mode'] = 'F'
+        else:
+            data['mode'] = 'unrecognized value'
+
+        speed_L = payload[0x0f]
+        speed_R = payload[0x10] 
+        if speed_L == 0x60 and speed_R == 0x00:
+            data['speed'] = 'L'
+        elif speed_L == 0x40 and speed_R == 0x00:
+            data['speed'] = 'M'
+        elif speed_L == 0x20 and speed_R == 0x00:
+            data['speed'] = 'H'
+        elif speed_L == 0x40 and speed_R == 0x80:
+            data['speed'] = 'Mu'
+        elif speed_R == 0x40:
+            data['speed'] = 'T'
+        elif speed_L == 0xa0 and speed_R == 0x00:
+            data['speed'] = 'A'
+        else:
+            data['speed'] = 'unrecognized value'
+
+        data['sleep'] = True if (payload[0x1a] & 0b100 == 0b000) else False
+        data['display'] = (payload[0x16] & 0x10 == 0x10)
+        data['health'] = (payload[0x14] & 0b11 == 0b11)
+
+        checksum = self._calculate_checksum(payload[:0x19], GET_STATE_CHECKSUM_TARGET) # checksum=(payload[0x1a] << 8) + payload[0x19]
+
+        if (payload[0x19] == checksum[0] and payload[0x1a] == checksum[1]):
+            pass # success
+        else:
+            print('checksum fail', ['{:02x}'.format(x) for x in checksum])
+
+        if (payload):
+            data['payload'] = payload
+
+        return data
+
+    def get_ac_info(self, payload=None):
+        """Returns dictionary with A/C info...
+        Not implemented yet, except power state.
+        """
+        payload = self._send_short_payload(0)
+
+        # first 13 bytes are the same: 22 00 bb 00 07 00 00 00 18 00 01 21 c0
+        data = {}
+        data['state'] = True if (payload[0x0d] & 0b1 == 0b1) else False
+
+        if (payload):
+            data['payload'] = payload
+
+        return data
+
+    def set_advanced(self, state=None, mode=None, set_temp=None, speed=None, swing_v=None, swing_h=None, sleep=None, display=None, health=None):
```

Comment:
Please use function annotations to make more clear what is the expected input and output.